keyboards/splitography/keymaps/beaklmu/common/keycode_functions.h

```diff
@@ -0,0 +1,660 @@
+
+// .......................................................... Keycode Primitives
+
+// register simple key press
+void tap_key(uint16_t keycode)
+{
+  register_code  (keycode);
+  unregister_code(keycode);
+}
+
+void shift_key(uint16_t keycode)
+{
+  register_code  (KC_LSFT);
+  tap_key        (keycode);
+  unregister_code(KC_LSFT);
+}
+
+#define SHIFT   1
+#define NOSHIFT 0
+
+static uint16_t key_timer = 0;
+
+// key press for thumb_layer() and lt_shift() macros
+bool key_press(uint8_t shift, uint16_t keycode)
+{
+  if (keycode) {
+    if (timer_elapsed(key_timer) < TAPPING_TERM) {
+      if (shift) {
+        shift_key(keycode);
+      }
+      else {
+        tap_key(keycode);
+      }
+      return true;
+    }
+  }
+  return false;
+}
```

Comment:
This looks like it's replicating the "Auto Shift" function? 

https://docs.qmk.fm/#/feature_auto_shift